src/Renderer/NavigatorRenderer.cpp

```diff
@@ -0,0 +1,378 @@
+  polyline_frame[9].y = polyline_frame[2].y; //
+  polyline_frame[10].x = polyline_frame[0].x; //
+  polyline_frame[10].y = polyline_frame[0].y; //
+  
+  canvas.Select(look.background_pen);
+  canvas.Select(look.background_brush);
+  canvas.DrawPolygon( polyline_frame, 11);
+
+  canvas.Select(look.frame_pen);
+  canvas.Select(look.frame_brush);
+  canvas.DrawPolyline( polyline_frame, 11);
+}
+
+
+void 
+NavigatorRenderer::DrawText(Canvas &canvas, const Waypoint &wp_current, 
+                                const PixelRect &rc, const NavigatorLook &look) {
+
+  const auto &basic = CommonInterface::Basic();
+  const auto &calculated = CommonInterface::Calculated();
+  TCHAR current_speed[256] = "---";
+
+  if (basic.ground_speed_available)
+    FormatSpeed(current_speed, Units::ToSysUnit(basic.ground_speed, Unit::METER_PER_SECOND),
+              Unit::KILOMETER_PER_HOUR, true, true);
+  
+  auto current_speed_s = static_cast<std::string>(current_speed);
+    
+  TextRenderer text_renderer1;
+  text_renderer1.SetVCenter(true);
+  text_renderer1.SetCenter(false);
+  text_renderer1.SetControl(true);
+
+  TextRenderer text_renderer2;
+  text_renderer2.SetVCenter(true);
+  text_renderer2.SetCenter(true);
+  text_renderer2.SetControl(true);
+  
+  canvas.Select(*look.font);
+
+  bool has_started = calculated.ordered_task_stats.start.task_started;
+
+  const auto time_elapsed = TimeStamp{FloatDuration{calculated.ordered_task_stats.total.time_elapsed}};
+  const RoughTimeDelta r{};
+
+  auto time_elapsed_s = FormatLocalTimeHHMM(time_elapsed, r).c_str();
+  has_started ? time_elapsed_s : time_elapsed_s = "--:--";
+
+  const auto time_start = calculated.ordered_task_stats.start.time;
+  auto time_start_s = FormatLocalTimeHHMM(time_start,
+                                  CommonInterface::GetComputerSettings().utc_offset).c_str();
+  has_started ? time_start_s : time_start_s = "--:--";
+
+  const auto time_local = FormatLocalTimeHHMM(basic.time,
+                                  CommonInterface::GetComputerSettings().utc_offset).c_str();
+
+  const auto time_planned = TimeStamp{FloatDuration{calculated.ordered_task_stats.total.time_planned}};
+  auto time_planned_s = FormatLocalTimeHHMM(time_planned, r).c_str();
+  has_started ? time_planned_s : time_planned_s = "--:--";
+
+  const auto arrival_planned = TimeStamp{FloatDuration{time_start.ToDuration() + time_planned.ToDuration()}};
+  auto arrival_planned_s = FormatLocalTimeHHMM(arrival_planned, 
+                                  CommonInterface::GetComputerSettings().utc_offset).c_str(); 
+  has_started ? arrival_planned_s : arrival_planned_s = "--:--";
+
+  auto waypoint_s = static_cast<std::string>(wp_current.name);
+
+  // WP Dist
+  const auto waypoint_distance = calculated.ordered_task_stats.current_leg.vector_remaining.distance;
+
+  std::ostringstream oss;
+  oss << std::fixed << std::setprecision(2) << std::noshowpoint  << waypoint_distance/1000; 
```

Comment:
Please please avoid using C++ stdstreams.

Reply:
ok, amended as follows:
```
// e_WP_Distance
TCHAR waypoint_distance_s[10]{};
const auto waypoint_distance {calculated.ordered_task_stats.current_leg.vector_remaining.distance};

FormatUserDistance(waypoint_distance,
                                  waypoint_distance_s, true, 0);
```

Reply:
Why do you insist on initialising all those buffers? What's the point? This is code bloat and runtime overhead. Not much, but completely unnecessary.

Reply:
That is because, I learned this way: I always initialized a new object (following the CppCoreGuideline / [ES.20: Always initialize an object](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es20-always-initialize-an-object))

I must admit, it's the first time that I initialized a char array. 
I did this mistake after an [interesting lecture](https://stackoverflow.com/questions/21946447/how-much-performance-difference-when-using-string-vs-char-array) of a comparison of efficiency between char array and std::string. the author adds this in the code that actually, you're right is code bloat and runtime overhead. adding an initialization to a char array, fill the array with '/0'. It's 'zeroing out memory'. I think it's not relevant in this situation...

--> initialization deleted

Reply:
This part of the CppCoreGuideline is so easily misunderstood that I consider it wrong; it's bad advice. Of course uninitialized variables often lead to bug, but adding arbitrary initializers is even more buggy, because it just hides the actual bug, instead of fixing it. I mean, what use is initializing the array with all zeroes? (That's a serious question.) What do these zeroes mean, and what would be wrong if the zeroes weren't there? Also a serious question. The only meaningful answer is: those zeroes do not mean anything, it's just an initializer for the sake of having an initializer, compliance theater instead of good code.